File: python3/easy/0009.py
class Solution:
    def isPalindrome(self, n):
        """(n: int) -> bool"""
        x, y = n, 0
        f = lambda: (y * 10) + x % 10
        while x > 0:
            x, y = x//10, f()
        return y == n


solution = Solution()

print(solution.isPalindrome(12345))
print(solution.isPalindrome(12321))

File: javascript/easy/0009.js
const isPalindrome = x => {
    if (x < 0 || (x % 10 === 0 && x !==0)) {
        return false
    }
    
    let reverse = 0;
    let j = x;
    
    while (j > 0){
        reverse = (reverse * 10) + (j % 10);
        j = ~~(j/10)
    }
    
    return x === reverse;
    
};